### ISSUE TITLE
version 3.5.1 KeyError crash 

### ISSUE THREAD

**Issue by scottnonnenberg** [elastic/curator#594]
This is a strange issue, because it seems like it never happens when I run it manually. But when I run it via `cron`, with stdout redirected to `/dev/null`, it crashes.

Version:

```
$ /usr/local/bin/curator --version
curator, version 3.5.1
```

Original command, run via `cron`:

```
/usr/local/bin/curator close indices --older-than 30 --time-unit days --timestring '%Y.%m.%d' > /dev/null
```

Full traceback:

```
Traceback (most recent call last):
 File "/usr/local/bin/curator", line 11, in <module>
   sys.exit(main())
 File "/usr/local/lib/python2.7/dist-packages/curator/curator.py", line 5, in main
   cli( obj={ "filters": [] } )
 File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 716, in __call__
   return self.main(*args, **kwargs)
 File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 696, in main
   rv = self.invoke(ctx)
 File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1060, in invoke
   return _process_result(sub_ctx.command.invoke(sub_ctx))
 File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1060, in invoke
   return _process_result(sub_ctx.command.invoke(sub_ctx))
 File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 889, in invoke
   return ctx.invoke(self.callback, **ctx.params)
 File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 534, in invoke
   return callback(*args, **kwargs)
 File "/usr/local/lib/python2.7/dist-packages/click/decorators.py", line 17, in new_func
   return f(get_current_context(), *args, **kwargs)
 File "/usr/local/lib/python2.7/dist-packages/curator/cli/index_selection.py", line 172, in indices
   retval = do_command(client, ctx.parent.info_name, working_list, ctx.parent.params, master_timeout)
 File "/usr/local/lib/python2.7/dist-packages/curator/cli/utils.py", line 233, in do_command
   return close(client, indices)
 File "/usr/local/lib/python2.7/dist-packages/curator/api/close.py", line 38, in close
   return close_indices(client, indices)
 File "/usr/local/lib/python2.7/dist-packages/curator/api/close.py", line 19, in close_indices
   seal_indices(client, indices)
 File "/usr/local/lib/python2.7/dist-packages/curator/api/seal.py", line 17, in seal_indices
   indices = prune_closed(client, indices)
 File "/usr/local/lib/python2.7/dist-packages/curator/api/utils.py", line 478, in prune_closed
   return prune_open_or_closed(client, indices, append_closed=False)
 File "/usr/local/lib/python2.7/dist-packages/curator/api/utils.py", line 495, in prune_open_or_closed
   if index_closed(client, idx) is append_closed:
 File "/usr/local/lib/python2.7/dist-packages/curator/api/utils.py", line 98, in index_closed
   return index_metadata['metadata']['indices'][index_name]['state'] == 'close'
KeyError: u'logstash-2016.01.19'
```

**Comment by untergeek** [comment]
There are two potential answers.

1. Escape the percent signs: https://github.com/elastic/curator/issues/151#issuecomment-88252653
2. If systemd, double the percents: https://discuss.elastic.co/t/curator-how-to-escape-characters-of-timestring-in-systemd/25222

**Comment by scottnonnenberg** [comment]
This is my crontab entry:

```
0 11 * * * /usr/local/bin/curator close indices --older-than 30 --time-unit days --timestring '\%Y.\%m.\%d' > /dev/null
```

**Comment by untergeek** [comment]
Yes, but is it systemd?  If so, it would need to have double percents instead.

Other alternatives are to use a wrapper script and call that in cron instead.  Some cron variants choke on the length of the command line.

**Comment by scottnonnenberg** [comment]
It's ubuntu 14.04. I'll try the double percent format for tonight's run.

**Comment by untergeek** [comment]
I recommend trying it sooner with the `--dry-run` flag (in your cron line) before just hoping it works for tonight's run.

**Comment by scottnonnenberg** [comment]
Actually, I just looked at my setup and I have a `curator delete` command in my crontab as well. Now I suspect that the delete is still propagating through elasticsearch when the close command tries to resolve. I'll put a five-minute delay between them.

**Comment by untergeek** [comment]
This should be addressed by #595 and #596 with Curator 4.0 imminent.

If you feel this is in error, please feel free to reopen this ticket or open another.
